### DISCUSSION TITLE
grantConsent() not loading

### DISCUSSION THREAD

**Discussion by urlexnilz** [johannschopplich/nuxt-gtag#45] [General]
Hi,
we are currently integrating nuxt-gtag into our consent management tool. 

Because GA cant be loaded without the proper permissions my initial config is:

```
	gtag: {
		id: 'G-XYZ',
		initialConsent: false,
	},
```

When I remove the line "initialConsent", GTAG is loaded, without any given consent. Now I'm listening for the event when the user gives his consent (which is working pretty fine). If the user grants the permission to load GA I want to enable GA. 

I'm using the code snippet that you provided in docs:
```
	const { grantConsent } = useGtag();
	... check if consent is given and if yes:
	grantConsent(); 
```

But nothing happens. I would assume GTAG is loaded after giving Consent, but I'm not seeing any calls to GA. I also tried to pass my GA ID in grantConsent(), but it doesn't change anything. 

```
	const { grantConsent } = useGtag();
	... check if consent is given and if yes:
	grantConsent('G-XYZ'); 
```

Am I missing something? 

Cheers

**Comment by johannschopplich** [comment]
Hi there! Nuxt Gtag should work as expected for your application. Did you make sure you ran `grantConsent` in the client? It won't have an effect on the server.

You can try the [playground](https://nuxt-gtag.netlify.app/) to see it in action.

Please open an issue with a reproducible example, thanks!

**Reply by urlexnilz** [reply]
Hi @johannschopplich :) 
thanks for your feedback; I've put my code in app.vue. This shouldnt make any difference, right? Or do I have to put it somewhere else?

**Reply by johannschopplich** [reply]
That's fine. 👍
